Index: framework/widget/dtd/widget-screen.xsd
===================================================================
--- framework/widget/dtd/widget-screen.xsd	(révision 1863538)
+++ framework/widget/dtd/widget-screen.xsd	(copie de travail)
@@ -411,6 +411,8 @@
     <xs:element name="include-portal-page" substitutionGroup="AllWidgets">
         <xs:complexType>
             <xs:attribute type="xs:string" name="id" />
+            <xs:attribute type="xs:string" name="portletId" />
+            <xs:attribute type="xs:string" name="portletSeqId" />
             <xs:attribute name="conf-mode" type="xs:boolean" default="false">
                 <xs:annotation>
                     <xs:documentation>Show the portal in configuration mode.</xs:documentation>
@@ -520,6 +522,12 @@
     <xs:attributeGroup name="attlist.html-template">
         <xs:attribute type="xs:string" name="location" use="required" />
     </xs:attributeGroup>
+    <xs:element name="vuejs" substitutionGroup="HtmlWidgets">
+        <xs:complexType>
+            <xs:attribute type="xs:string" name="component-name" use="required" />
+            <xs:attribute type="xs:string" name="param-map" />
+        </xs:complexType>
+    </xs:element>
     <xs:element name="html-template-decorator" substitutionGroup="HtmlWidgets">
         <xs:annotation>
             <xs:documentation>
