Index: framework/widget/dtd/widget-common.xsd
===================================================================
--- framework/widget/dtd/widget-common.xsd	(révision 1853222)
+++ framework/widget/dtd/widget-common.xsd	(copie de travail)
@@ -515,6 +515,13 @@
             <xs:attribute name="send-if-empty" type="xs:boolean" default="true"/>
         </xs:complexType>
     </xs:element>
+    <xs:element name="auto-parameters-portlet">
+        <xs:annotation>
+            <xs:documentation>Generate 3 parameters : portalPageId, portalPortletId and portletSeqId update from-field parameters.xxx</xs:documentation>
+        </xs:annotation>
+        <xs:complexType>
+        </xs:complexType>
+    </xs:element>
     <xs:element name="exclude">
         <xs:complexType>
             <xs:attribute name="field-name" type="xs:string" use="required" />
@@ -525,6 +532,7 @@
             <xs:element minOccurs="0" ref="auto-parameters-service"/>
             <xs:element minOccurs="0" ref="auto-parameters-entity"/>
             <xs:element minOccurs="0" maxOccurs="unbounded" ref="parameter" />
+            <xs:element minOccurs="0" ref="auto-parameters-portlet"/>
             <xs:element minOccurs="0" name="image" type="image" />
         </xs:sequence>
         <xs:attribute type="xs:string" name="text" />
@@ -559,6 +567,20 @@
                     <xs:enumeration value="anchor" />
                     <xs:enumeration value="hidden-form" />
                     <xs:enumeration value="layered-modal" />
+                    <xs:enumeration value="set-area" >
+                        <xs:annotation>
+                            <xs:documentation>
+                                Usable only in menu link, for FrontJs action, target and target-windows are used.
+                            </xs:documentation>
+                        </xs:annotation>
+                    </xs:enumeration>
+                    <xs:enumeration value="set-watcher" >
+                        <xs:annotation>
+                            <xs:documentation>
+                                Usable only in menu link, for FrontJs action, only target-windows is used (as watcherName).
+                            </xs:documentation>
+                        </xs:annotation>
+                    </xs:enumeration>
                 </xs:restriction>
             </xs:simpleType>
         </xs:attribute>
