Index: framework/widget/src/main/java/org/apache/ofbiz/widget/model/CommonWidgetModels.java
===================================================================
--- framework/widget/src/main/java/org/apache/ofbiz/widget/model/CommonWidgetModels.java	(révision 1870471)
+++ framework/widget/src/main/java/org/apache/ofbiz/widget/model/CommonWidgetModels.java	(copie de travail)
@@ -364,14 +364,22 @@
                 this.linkType = linkElement.getAttribute("link-type");
             }
             List<? extends Element> parameterElementList = UtilXml.childElementList(linkElement, "parameter");
-            if (parameterElementList.isEmpty()) {
+            boolean autoPortletParamsElement = UtilXml.firstChildElement(linkElement, "auto-parameters-portlet") == null ? false : true;
+            if (parameterElementList.isEmpty() && ! autoPortletParamsElement) {
                 this.parameterList = Collections.emptyList();
             } else {
+                int paramListSize = parameterElementList.size() + (autoPortletParamsElement ? 4 : 0);
                 List<Parameter> parameterList = new ArrayList<>(
-                        parameterElementList.size());
+                        paramListSize);
                 for (Element parameterElement : parameterElementList) {
                     parameterList.add(new Parameter(parameterElement));
                 }
+                if (autoPortletParamsElement) {
+                    parameterList.add(new CommonWidgetModels.Parameter("portalPageId",    "parameters.portalPageId",    true));
+                    parameterList.add(new CommonWidgetModels.Parameter("portalPortletId", "parameters.portalPortletId", true));
+                    parameterList.add(new CommonWidgetModels.Parameter("portletSeqId",    "parameters.portletSeqId",    true));
+                    parameterList.add(new CommonWidgetModels.Parameter("currentAreaId",   "parameters.currentAreaId",   true));
+                }
                 this.parameterList = Collections.unmodifiableList(parameterList);
             }
             Element autoServiceParamsElement = UtilXml.firstChildElement(linkElement, "auto-parameters-service");
