Index: framework/widget/src/main/java/org/apache/ofbiz/widget/model/ModelWidgetVisitor.java
===================================================================
--- framework/widget/src/main/java/org/apache/ofbiz/widget/model/ModelWidgetVisitor.java	(révision 1870471)
+++ framework/widget/src/main/java/org/apache/ofbiz/widget/model/ModelWidgetVisitor.java	(copie de travail)
@@ -68,6 +68,8 @@
 
     void visit(ModelScreenWidget.Label label) throws Exception;
 
+    void visit(ModelScreenWidget.VueJs vuejs) throws Exception;
+
     void visit(ModelScreenWidget.ScreenLink link) throws Exception;
 
     void visit(ModelScreenWidget.Menu menu) throws Exception;
