Index: framework/widget/src/main/java/org/apache/ofbiz/widget/model/ModelForm.java
===================================================================
--- framework/widget/src/main/java/org/apache/ofbiz/widget/model/ModelForm.java	(révision 1853222)
+++ framework/widget/src/main/java/org/apache/ofbiz/widget/model/ModelForm.java	(copie de travail)
@@ -432,7 +432,11 @@
                 } else {
                     onPaginateUpdateAreas.add(updateArea);
                 }
-            } else if ("submit".equals(updateArea.getEventType())) {
+            } else if ("submit".equals(updateArea.getEventType())
+                     ||"updateStore".equals(updateArea.getEventType())
+                     ||"post-updateStore".equals(updateArea.getEventType())
+                     ||"post-setArea".equals(updateArea.getEventType())
+                      ) {
                 int index = onSubmitUpdateAreas.indexOf(updateArea);
                 if (index != -1) {
                     onSubmitUpdateAreas.set(index, updateArea);
