Index: framework/widget/src/main/java/org/apache/ofbiz/widget/model/ModelForm.java
===================================================================
--- framework/widget/src/main/java/org/apache/ofbiz/widget/model/ModelForm.java	(révision 1861341)
+++ framework/widget/src/main/java/org/apache/ofbiz/widget/model/ModelForm.java	(copie de travail)
@@ -432,7 +432,13 @@
                 } else {
                     onPaginateUpdateAreas.add(updateArea);
                 }
-            } else if ("submit".equals(updateArea.getEventType())) {
+            } else if ("submit".equals(updateArea.getEventType())
+                     ||"post".equals(updateArea.getEventType())
+                     ||"setArea".equals(updateArea.getEventType())
+                     ||"setWatcher".equals(updateArea.getEventType())
+                     ||"collapse".equals(updateArea.getEventType())
+                     ||"closeModal".equals(updateArea.getEventType())
+                      ) {
                 int index = onSubmitUpdateAreas.indexOf(updateArea);
                 if (index != -1) {
                     onSubmitUpdateAreas.set(index, updateArea);
@@ -1713,13 +1719,21 @@
             this.defaultServiceName = defaultServiceName;
             this.defaultEntityName = defaultEntityName;
             List<? extends Element> parameterElementList = UtilXml.childElementList(updateAreaElement, "parameter");
-            if (parameterElementList.isEmpty()) {
+            boolean autoPortletParamsElement = UtilXml.firstChildElement(updateAreaElement, "auto-parameters-portlet") == null ? false : true;
+            if (parameterElementList.isEmpty() && ! autoPortletParamsElement) {
                 this.parameterList = Collections.emptyList();
             } else {
-                List<CommonWidgetModels.Parameter> parameterList = new ArrayList<>(parameterElementList.size());
+                int paramListSize = parameterElementList.size() + (autoPortletParamsElement ? 4 : 0);
+                List<CommonWidgetModels.Parameter> parameterList = new ArrayList<>(paramListSize);
                 for (Element parameterElement : parameterElementList) {
                     parameterList.add(new CommonWidgetModels.Parameter(parameterElement));
                 }
+                if (autoPortletParamsElement) {
+                    parameterList.add(new CommonWidgetModels.Parameter("portalPageId",    "parameters.portalPageId",    true));
+                    parameterList.add(new CommonWidgetModels.Parameter("portalPortletId", "parameters.portalPortletId", true));
+                    parameterList.add(new CommonWidgetModels.Parameter("portletSeqId",    "parameters.portletSeqId",    true));
+                    parameterList.add(new CommonWidgetModels.Parameter("currentAreaId",   "parameters.currentAreaId",   true));
+                }
                 this.parameterList = Collections.unmodifiableList(parameterList);
             }
             Element autoServiceParamsElement = UtilXml.firstChildElement(updateAreaElement, "auto-parameters-service");
@@ -1760,6 +1774,10 @@
             return areaId;
         }
 
+        public String getAreaId(Map<String, ? extends Object> context) {
+            return FlexibleStringExpander.expandString(areaId, context);
+        }
+
         public String getAreaTarget(Map<String, ? extends Object> context) {
             return FlexibleStringExpander.expandString(areaTarget, context);
         }
@@ -1811,5 +1829,14 @@
         public List<CommonWidgetModels.Parameter> getParameterList() {
             return parameterList;
         }
+
+        public Map<String, Object> toMap(Map<String, Object> context) {
+            Map<String, Object> map = new HashMap<>();
+            map.put("eventType",    this.getEventType());
+            map.put("areaId",       FlexibleStringExpander.expandString(this.getAreaId(), context));
+            map.put("areaTarget",   FlexibleStringExpander.expandString(this.getAreaTarget(), context));
+            map.put("parameterMap", this.getParameterMap(context));
+            return map;
+        }
     }
 }
