diff --git a/framework/widget/src/main/java/org/apache/ofbiz/widget/model/XmlWidgetVisitor.java b/framework/widget/src/main/java/org/apache/ofbiz/widget/model/XmlWidgetVisitor.java
index 0777a7058e..c7ecc9fa64 100644
--- a/framework/widget/src/main/java/org/apache/ofbiz/widget/model/XmlWidgetVisitor.java
+++ b/framework/widget/src/main/java/org/apache/ofbiz/widget/model/XmlWidgetVisitor.java
@@ -44,6 +44,7 @@ import org.apache.ofbiz.widget.model.ModelScreenWidget.ScreenLink;
 import org.apache.ofbiz.widget.model.ModelScreenWidget.Screenlet;
 import org.apache.ofbiz.widget.model.ModelScreenWidget.Section;
 import org.apache.ofbiz.widget.model.ModelScreenWidget.Tree;
+import org.apache.ofbiz.widget.model.ModelScreenWidget.VueJs;
 import org.apache.ofbiz.widget.model.ModelTree.ModelNode;
 import org.apache.ofbiz.widget.model.ModelTree.ModelNode.ModelSubNode;
 
@@ -230,6 +231,16 @@ public class XmlWidgetVisitor extends XmlAbstractWidgetVisitor implements ModelW
     }
 
     @Override
+    public void visit(VueJs vuejs) throws Exception {
+        writer.append("<vuejs");
+        visitModelWidget(vuejs);
+        visitAttribute("component-name", vuejs.getComponentNameExdr());
+        writer.append(">");
+        visitParameters(vuejs.getParameterList());
+        writer.append("</label>");
+    }
+
+    @Override
     public void visit(Menu menu) throws Exception {
         writer.append("<include-menu");
         visitModelWidget(menu);
