Index: framework/widget/src/main/java/org/apache/ofbiz/widget/renderer/macro/MacroScreenRenderer.java
===================================================================
--- framework/widget/src/main/java/org/apache/ofbiz/widget/renderer/macro/MacroScreenRenderer.java	(révision 1863538)
+++ framework/widget/src/main/java/org/apache/ofbiz/widget/renderer/macro/MacroScreenRenderer.java	(copie de travail)
@@ -223,6 +223,10 @@
         parameters.put("style", label.getStyle(context));
         executeMacro(writer, "renderLabel", parameters);
     }
+    @Override
+    public void renderVueJs(Appendable writer, Map<String, Object> context, ModelScreenWidget.VueJs vuejs) throws IOException {
+        //TODO should be implemented
+    }
 
     @Override
     public void renderHorizontalSeparator(Appendable writer, Map<String, Object> context, ModelScreenWidget.HorizontalSeparator separator) throws IOException {
