Index: framework/widget/src/main/java/org/apache/ofbiz/widget/renderer/ScreenStringRenderer.java
===================================================================
--- framework/widget/src/main/java/org/apache/ofbiz/widget/renderer/ScreenStringRenderer.java	(révision 1861341)
+++ framework/widget/src/main/java/org/apache/ofbiz/widget/renderer/ScreenStringRenderer.java	(copie de travail)
@@ -46,6 +46,7 @@
 
     public void renderHorizontalSeparator(Appendable writer, Map<String, Object> context, ModelScreenWidget.HorizontalSeparator separator) throws IOException;
     public void renderLabel(Appendable writer, Map<String, Object> context, ModelScreenWidget.Label label) throws IOException;
+    public void renderVueJs(Appendable writer, Map<String, Object> context, ModelScreenWidget.VueJs vuejs) throws IOException;
     public void renderLink(Appendable writer, Map<String, Object> context, ModelScreenWidget.ScreenLink link) throws IOException;
     public void renderImage(Appendable writer, Map<String, Object> context, ModelScreenWidget.ScreenImage image) throws IOException;
 
